Assets/Code/Scripts/TutorialManagement/TutorialController.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using POLIMIGameCollective;
+using TMPro;
+using Unity.VisualScripting;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class TutorialController : MonoBehaviour
+{
+    [SerializeField] private GameObject _controlsContainer;
+    [SerializeField] private TMP_Text _tutorialText;
+    [SerializeField] private float _hintDuration;
+    private float _endHintTime;
+
+    private void Awake()
+    {
+        EventManager.StartListening ("PressP", PressP);
+        EventManager.StartListening ("DisplayHint", DisplayHint);
+
+    }
+
+
+    private void PressP()
+    {
+        EventManager.StopListening("PressP", PressP);
+        
+        Debug.Log("PressP displayed");
+        
+        EventManager.StartListening("PressP", PressP);
+    }
```

Comment:
At the end we did not use this function, feel free to keep it just to show an example of an event without inputs from the trigger. If you keep it, I would refactor it as follows
```suggestion
    private void Awake()
    {
        EventManager.StartListening ("PressP", PressP);
        EventManager.StartListening ("DisplayHint", DisplayHint);
    }


    private void PressP()
    {
        EventManager.StopListening("PressP", PressP);
        StartCoroutine(DisplayHintCoroutine("PRESS P"));
        EventManager.StartListening("PressP", PressP);
    }
```